dotnet-authserver/src/TeacherIdentity.AuthServer/AuthenticationState.cs

```diff
@@ -106,6 +106,12 @@ public class AuthenticationState
     public string? MobileNumber { get; private set; }
     [JsonInclude]
     public bool MobileNumberVerified { get; private set; }
+    [JsonInclude]
+    public Guid? ExistingAccountUserId { get; private set; }
+    [JsonInclude]
+    public string? ExistingAccountEmail { get; private set; }
+    [JsonInclude]
+    public bool? ExistingAccountConfirmed { get; private set; }
```

Comment:
Implemented slightly differently from the ticket. We needed a way to pass ExistingAccountEmail to the check-account page without exposing it to the browser.

Reply:
Not sure if ExistingAccountConfirmed property interferes with what you had in mind for the next page

Reply:
I wonder if we could use a different term to 'Confirmed' here; I initially read this as meaning the user has verified they can access the existing account. Perhaps `ExistingAccountChosen`?